internal/dcache/replication_manager/replication_manager.go

```diff
@@ -464,12 +621,13 @@ retry:
 
 			continue
 		}
+		log.Err("ReplicationManager::WriteMV: PutChunk to %s/%s failed [%v]",
+			respItem.rvName, req.MvName, respItem.err)
 
-		log.Err("ReplicationManager::WriteMV: PutChunk to %s/%s, node %s failed [%v]",
-			respItem.rvName, req.MvName, respItem.targetNodeID, respItem.err)
+		common.Assert(respItem.putChunkResp == nil)
 
 		rpcErr := rpc.GetRPCResponseError(respItem.err)
-		if rpcErr == nil {
+		if rpcErr == nil || rpcErr.GetCode() == models.ErrorCode_ThriftError {
```

Comment:
once we make PutChunkResponseOrError.Error as type error, then we don't need this special check.

Reply:
Same as above comment.